values.yaml

```diff
@@ -1,402 +1,134 @@
-  enable: true
-  # ingress.useDefault -- If set to true, will deploy an nginx ingress that will
-  # allow you to access Coder from an external IP address, but only if your
-  # kubernetes cluster is configured to provision external IP addresses. If you
-  # would like to bring your own ingress and hook Coder into that instead, set
-  # this value to false.
-  useDefault: true
-  # ingress.usePathWildcards -- Whether or not the ingress object should use
-  # path wildcards, i.e., ending with "/*". Some ingresses require this
-  # while others do not. You should check which path style your ingress
-  # requires. For ingress-nginx this should be set to false.
-  usePathWildcards: false
-  # ingress.host -- The hostname to use for accessing the platform. This can
-  # be left blank, and the user can still access the platform from the external
-  # IP or a DNS name that resolves to the external IP address.
-  host: ""
-  # ingress.loadBalancerIP -- sets the external IP address of the ingress to the
-  # provided value.
-  loadBalancerIP: ""
-  # ingress.loadBalancerSourceRanges -- If specified and supported by the
-  # platform, the traffic through the cloud-provider load-balancer will be
-  # restricted to the specified client IPs. This field will be ignored if the
-  # cloud-provider does not support the feature.
-  loadBalancerSourceRanges: []
-  # ingress.podSecurityPolicyName -- The name of the pod security policy the
-  # built-in ingress controller should abide. It should be noted that the
-  # ingress controller requires the `NET_BIND_SERVICE` capability, privilege
-  # escalation, and access to privileged ports to successfully deploy. Ignored
-  # if `ingress.useDefault` is false.
-  podSecurityPolicyName: ""
-  # ingress.additionalAnnotations -- Deprecated. Please use `ingress.annotations`.
-  additionalAnnotations: []
-  # ingress.annotations -- Additional annotations to be used when creating the
-  # ingress. These only apply to the Ingress Kubernetes kind. The annotations
-  # can be used to specify certificate issuers or other cloud provider-specific
-  # integrations.
-  annotations: {}
-  # ingress.tls -- TLS options for the ingress. The hosts used for the tls
-  # configuration come from the ingress.host and the devurls.host variables. If
-  # those don't exist, then the TLS configuration will be ignored.
+# coderd -- Primary service responsible for all things Coder!
+coderd:
+  # coderd.image -- Injected by Coder during release.
+  image: ""
+  # coderd.replicas -- The number of Kubernetes Pod replicas.
+  replicas: 1
+  # coderd.serviceSpec -- Specification to inject for the coderd service. See:
+  # https://kubernetes.io/docs/concepts/services-networking/service/
+  serviceSpec:
+    type: LoadBalancer
+    loadBalancerIP: ""
+    loadBalancerSourceRanges: []
+  # coderd.tls -- TLS configuration for coderd.
+  # These options will override dashboard configuration.
```

Comment:
You can't do any TLS config in the dash today, so maybe we hold off on this comment?

Reply:
I'd call this an end-state for 1.21. Since @lilshoff already did all the API stuff here, I'm confident we could do the UI by 1.21.